src/index.ts

```diff
@@ -99,6 +158,81 @@ export interface RegisterOptions extends CreateOptions {
   preferTsExts?: boolean | null
 }
 
+/*
+ * This interface exists solely for generating a JSON schema for tsconfig.json.
+ * We do *not* extend the compiler's tsconfig interface.  Instead we handle that
+ * on a schema level, via "allOf", so we pull in the same schema that VSCode
+ * already uses.
+ */
+/**
+ * tsconfig schema which includes "ts-node" options.
+ * @allOf [{"$ref": "https://schemastore.azurewebsites.net/schemas/json/tsconfig.json"}]
+ */
+export interface TsConfigSchema {
+  /**
+   * ts-node options.  See also: https://github.com/TypeStrong/ts-node#configuration-options
+   *
+   * ts-node offers TypeScript execution and REPL for node.js, with source map support.
+   */
+  'ts-node': TsConfigOptions
+}
+
+export interface TsConfigOptions
+  extends Omit<RegisterOptions,
+    | 'transformers'
+    | 'readFile'
+    | 'fileExists'
+    | 'skipProject'
+    | 'project'
+    | 'dir'
+  > {
+  requires?: Array<string>
```

Comment:
Why this this allowed? It's not used except via CLI. I think we should not include any option support that isn't part of `register`.

Reply:
Yeah, it could be confusing.  I was thinking there might be a situation where a project bootstraps the environment by convention, maybe to tweak the source-map-support or something, so they put ts-node in the shebang and expect it to `require()` a bootstrapper.  I guess I was thinking that ts-node is, itself, bootstrapping the node environment to do something special, so it makes sense to allow that bootstrapping to be extended.

I can remove it if it's a blocker.